README.md

```diff
@@ -38,19 +38,28 @@ Check out [`buildkite-elastic.yml`](templates/buildkite-elastic.yml) for more de
 
 Set your [Agent Query Rules](https://buildkite.com/docs/agent/agent-meta-data) to `queue=elastic`, or to whatever `BuildkiteQueue` you provided to your stack.
 
-## Credentials
+## Secrets
 
-Your stack has access to the `ProvisionBucket` parameter you passed in, you can use this to get a GitHub SSH key to the build and soon will be able to use it to get generic credentials there too.
+Your stack has access to the `SecretsBucket` parameter you passed in. This should be used in combination with strong encryption to ensure that your CI secrets (such as Github credentials) are reasonably secure. See the Security section for more details.
 
-For now, upload your GitHub key to the bucket, which should be private by default.
+You provide a key via an environment var in your Buildkite config called `BUILDKITE_SECRETS_KEY` which will be used to decrypt all the files found in the secrets bucket.
 
-For Docker Hub credentials, you can use `DOCKER_HUB_USER`, `DOCKER_HUB_PASSWORD` and `DOCKER_HUB_EMAIL`.
+Two files are specifically looked for, `id_rsa_github`, for checking out your git code and optionally `env`, which contains environment variables to expose to the job command.
+
+### Uploading your Secrets
 
 ```bash
-aws s3 cp --acl private my_id_rsa_key "s3://my-provision-bucket/id_rsa_github"
+PASSPHRASE=$(date +%s%N | sha256sum | base64 | head -c 32 ; echo) # generates a 32 character randomish password, use gdate for MacOS
```

Comment:
This is pretty terrible advice for key generation. If someone knows roughly when you create it they have a very small space to search - one per second they are off by

What about:
```
< /dev/urandom tr -dc '[:alnum:]' | head -c32
```